sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/key_client.hpp

```diff
@@ -51,12 +51,18 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
    * Vault. The client supports creating, retrieving, updating, deleting, purging, backing up,
    * restoring, and listing the KeyVaultKey.
    */
-  class KeyClient {
+  class KeyClient
+#if !defined(TESTING_BUILD)
+      final
+#endif
```

Comment:
mocked_transport_adapter_test.hpp
```
  // A derived class with no credential and authentication
  class KeyClientWithNoAuthenticationPolicy : public Azure::Security::KeyVault::Keys::KeyClient {
  public:
    explicit KeyClientWithNoAuthenticationPolicy(
        std::string const& vaultUrl,
        KeyClientOptions const& options = KeyClientOptions())
        : KeyClient(vaultUrl, nullptr, options)
    {
      auto apiVersion = options.Version.ToString();

      m_pipeline = std::make_unique<Azure::Security::KeyVault::_internal::KeyVaultPipeline>(
          Azure::Core::Url(vaultUrl),
          apiVersion,
          Azure::Core::Http::_internal::HttpPipeline(options, "test", "version", {}, {}));
    }
  };
```

telemetry_header_test.cpp
```
TEST_F(MockedTransportAdapterTest, keyvaultTelemetryId)
{
  std::string applicationId("ourApplicationId");
  m_clientOptions.Telemetry.ApplicationId = applicationId;
  m_client = std::make_unique<
      Azure::Security::KeyVault::Keys::Test::KeyClientWithNoAuthenticationPolicy>(
      "url", m_clientOptions);

  // The fake response from the mocked transport adapter is good for parsing a Key back
  auto response = m_client->GetKey("name");

  // The response is an echo of the sent headers. Let's find the telemetry id
  auto foundHeader = false;
  for (auto& header : response.RawResponse->GetHeaders())
  {
    if (Azure::Core::_internal::StringExtensions::LocaleInvariantCaseInsensitiveEqual(
            header.first, "User-Agent"))
    {
      foundHeader = true;
      EXPECT_PRED2(
          [](std::string const& received, std::string const& sent) {
            auto telemetryInfoWithNoOSAndDate = received.substr(0, sent.size());
            return Azure::Core::_internal::StringExtensions::LocaleInvariantCaseInsensitiveEqual(
                telemetryInfoWithNoOSAndDate, sent);
          },
          header.second,
          applicationId);
      break;
    }
  }
  EXPECT_TRUE(foundHeader);
}
```

This is the only place.

Reply:
> This is the only test that depended on it. I did not judge whether the test could get away without inheriting

Can you please file an issue for this. We may want to either
1) update the test to avoid the inheritance.
2) use this as a reason to leave keyclient as non-fnal.

cc @vhvb1989